event.go

```diff
@@ -414,81 +586,103 @@ const (
 // Returns an error if the total length of the event JSON is too long.
 // Returns an error if the event ID doesn't match the origin of the event.
 // https://matrix.org/docs/spec/client_server/r0.2.0.html#size-limits
-func (e Event) CheckFields() error { // nolint: gocyclo
+func (e *Event) CheckFields() error { // nolint: gocyclo
+	var fields eventFields
+	switch f := e.fields.(type) {
+	case eventFormatV1Fields:
+		if f.AuthEvents == nil || f.PrevEvents == nil {
+			return errors.New("gomatrixserverlib: auth events and prev events must not be nil")
+		}
+		fields = f.eventFields
+	case eventFormatV2Fields:
+		if f.AuthEvents == nil || f.PrevEvents == nil {
+			return errors.New("gomatrixserverlib: auth events and prev events must not be nil")
```

Comment:
Do we actually are about `nil` or do we care about zero length? No commentary to help me identify what your intent is here.